### PR TITLE
Return of the --open flag 

### PR DESCRIPTION
**Pull request by dalanmiller**
Re-added this since serve.js had deviated a bit from previous. Now works only with `--open` flag as desired from comments before. Only wish would be that in the case of multiple `--open`ings that it would just refresh the already opened window and tab.

https://stackoverflow.com/questions/8454510/open-url-in-same-window-and-in-same-tab

closes #223

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/504)

### CODE REVIEW

**Comment by deontologician** [comment]
:lgtm: I'll leave it to you to merge it, depending on whether you feel like switching to accessSync or not

<details><summary>

<i>Previously, dalanmiller (Daniel Alan Miller) wrote…</i></summary>



> #### Return of the --open flag
> 
> Re-added this since serve.js had deviated a bit from previous. Now works only with `--open` flag as desired from comments before. Only wish would be that in the case of multiple `--open`ings that it would just refresh the already opened window and tab.
> 
> https://stackoverflow.com/questions/8454510/open-url-in-same-window-and-in-same-tab
> 
> closes #223
> 
> </details>

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[cli/src/serve.js, line 584 [r1]](https://reviewable.io:443/reviews/rethinkdb/horizon/504#-KI_FVEvZAdveHFqgRlY:-KI_FVEvZAdveHFqgRlZ:743469302) ([raw file](https://github.com/rethinkdb/horizon/blob/000a7531e67b7033c7cf08cbe6e0bbc5ef88dfe8/cli/src/serve.js#L584)):_

> ``` JavaScript
>     if (opts.open && opts.serve_static) {
>       // Check if index.html exists and readable in serve static_static directory
>       fs.access(`${opts.serve_static}/index.html`, fs.R_OK | fs.F_OK, (err) => {
> ```

You could use accessSync here, we've been using the sync versions in the cli

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/rethinkdb/horizon/504#-:-KI_FesIS5OWlJICTqH7:1034430199)_

File: cli/src/serve.js [hunk 1]
```diff
@@ -565,6 +576,28 @@ const runCommand = (opts, done) => {
                                       Object.assign({}, { path: name }, opts.auth[name]));
       }
     }
+  }).then(() => {
+    // Automatically open up index.html in the `dist` directory only if
+    //  in dev mode and an index.html exists in the directory.
```
Comment (Tryneus) [inline cli/src/serve.js]: why does this mention dev mode?

Comment (dalanmiller) [inline cli/src/serve.js]: Originally, it opened the browser when in dev mode. I just forgot to take this comment out.

**Comment by dalanmiller** [comment]
@deontologician appropriate use of `accessSync`?

https://github.com/rethinkdb/horizon/pull/504/files#diff-7e8aa0fffc19f1f930c3f20aff511503R585

**Comment by deontologician** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/rethinkdb/horizon/504)_

**Pull request merged**